File: Programmers/숫자 문자열과 영단어.py
def solution(s):
    voca = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    voca_dict = {}
    for i in range(10):
        voca_dict[voca[i]] = str(i)
    temp = ""
    i = 0
    while i < len(s):
        if s[i].isdigit() == True:
            temp += s[i]
            i += 1
        else:
            check = ""
            current_idx = i
            while current_idx < len(s):
                if check in voca_dict:
                    break;
                if s[current_idx].isdigit() == False:
                    check += s[current_idx]
                    current_idx += 1
                else:
                    break;
            i = current_idx
            temp += voca_dict[check]

    answer = int(temp)
    return answer

File: Programmers JS/숫자 문자열과 영단어.js
function solution(s) {
  let numbers = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];
  var answer = s;

  for (let i = 0; i < numbers.length; i++) {
    let arr = answer.split(numbers[i]);
    console.log(arr);
    answer = arr.join(i);
    console.log(answer);
  }
  //     var answer = "";
  //     let numberMap = {
  //         zero: 0,
  //         one: 1,
  //         two:2,
  //         three:3,
  //         four:4,
  //         five:5,
  //         six:6,
  //         seven:7,
  //         eight:8,
  //         nine:9,
  //     }
  //     let temp = '';
  //     s = s.split("")

  //     while(s.length){

  //         let pop = s.shift();
  //         if(pop.match(/[0-9]/)){
  //             answer += pop;
  //         }else{
  //             temp += pop;
  //             if(temp in numberMap){
  //                 answer += numberMap[temp].toString();
  //                 temp = ""
  //             }
  //         }
  //     }
  return Number(answer);
}
